sdk/include/platform/sunburst/platform-spi.hh

```diff
@@ -0,0 +1,92 @@
+#pragma once
+#include <cdefs.h>
+#include <stdint.h>
+
+/**
+ * A Simple Driver for the Sonata's GPIO.
+ *
+ * Documentation source can be found at:
+ * https://github.com/lowRISC/sonata-system/blob/1a59633d2515d4fe186a07d53e49ff95c18d9bbf/doc/ip/spi.md
+ *
+ * Rendered documentation is served from:
+ * https://lowrisc.org/sonata-system/doc/ip/spi.html
+ */
+struct SonataSpi
+{
+	// Interrupt State Register
+	uint32_t intrState;
+	// Interrupt Enable Register
+	uint32_t intrEnable;
+	// Interrupt Test Register
+	uint32_t intrTest;
+	// Configuration register. Controls how the SPI block transmits
+	uint32_t cfg;
+	// Controls the operation of the SPI block. This register can
+	uint32_t control;
+	// Status information about the SPI block
+	uint32_t status;
+	// When written begins an SPI operation. Writes are ignored when the
+	uint32_t start;
+	// Data from the receive FIFO. When read the data is popped from the
+	uint32_t rxFifo;
+	// Bytes written here are pushed to the transmit FIFO. If the FIFO
+	uint32_t txFifo;
+
+
+	// Initialises the SPI block
+	void
+	init(bool cpol, bool cpha, bool msbFirst, uint16_t halfClkPeriod) volatile
+	{
+		cfg = (cpol ? 1 << 31 : 0) | (cpha ? 1 << 30 : 0) |
+		      (msbFirst ? 1 << 29 : 0) | halfClkPeriod;
+	}
+
+	// Waits for the SPI device to become idle
+	void wait_idle() volatile
+	{
+		// Wait whilst IDLE field in STATUS is low
+		while ((status & 0x40000) == 0) {}
+	}
+
+	// Sends `len` bytes from the given `data` buffer.
+	void tx(const uint8_t data[], uint32_t len) volatile
```

Comment:
It would be nice to use `std::span<uint8_t` but we don't have that in the CHERIoT standard library, so I resorted to simple C arrays.

Reply:
That’s fine (though please name these ‘transmit’ and ‘receive’ or similar). We probably should also have a concept covering this interface.